game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/air/AirAttackVsNonSubsStep.java

```diff
@@ -2,18 +2,20 @@
 
 import static games.strategy.triplea.delegate.battle.BattleStepStrings.AIR_ATTACK_NON_SUBS;
 
-import games.strategy.triplea.delegate.battle.BattleActions;
 import games.strategy.triplea.delegate.battle.BattleState;
 import java.util.List;
 
 /** Air can not attack subs unless a destroyer is present */
 public class AirAttackVsNonSubsStep extends AirVsNonSubsStep {
-  public AirAttackVsNonSubsStep(final BattleState battleState, final BattleActions battleActions) {
-    super(battleState, battleActions);
+  public AirAttackVsNonSubsStep(final BattleState battleState) {
+    super(battleState);
   }
 
   @Override
   public List<String> getNames() {
+    if (!valid()) {
```

Comment:
nit, can use a ternary here to be a bit more concise. It's generally a bit easier (lower cognitive load) for if statements to lead with the 'true' value.

IE:
```
return valid()
   ? List.of(AIR_ATTACK_NON_SUBS)
   : List.of();
```